lib/pengine/unpack.c

```diff
@@ -3657,6 +3658,29 @@ remap_monitor_rc(int rc, xmlNode *xml_op, const pe_node_t *node,
     return remapped_rc;
 }
 
+static void
+remap_operation(xmlNode *xml_op, pe_resource_t *rsc, pe_node_t *node,
+                pe_working_set_t *data_set, enum action_fail_response *on_fail,
+                int target_rc, int *rc, int *status) {
+    const char *task = crm_element_value(xml_op, XML_LRM_ATTR_TASK);
+    const char *task_key = get_op_key(xml_op);
+
+    if (pcmk__str_eq(task, CRMD_ACTION_STATUS, pcmk__str_none)) {
+        *rc = remap_monitor_rc(*rc, xml_op, node, rsc, data_set);
+    }
+
+    /* If the executor reported an operation status of anything but done or
+     * error, consider that final. But for done or error, we know better whether
+     * it should be treated as a failure or not, because we know the expected
+     * result.
+     */
+    if (*status == PCMK_EXEC_DONE || *status == PCMK_EXEC_ERROR) {
+        *status = remap_op_status(rsc, *rc, target_rc, node, xml_op,
+                                      on_fail, data_set);
+        pe_rsc_trace(rsc, "Remapped %s status to %d", task_key, *status);
+    }
```

Comment:
I was thinking if we actually consolidate the two functions, it will simplify the later code (which needs to remap both rc and status together)

Reply:
Yeah, I figured that was what you meant but thought I'd try it as separate functions first because of how involved remap_op_status is.  I've changed it to just be one big function.